[Bug] PAX build error in REL_2_STABLE

### Apache Cloudberry version

REL_2_STABLE

### What happened

When running `make -j$(nproc) -C ~/cloudberry`, errors returned:

```
build pax, USE_MANIFEST_API=OFF USE_PAX_CATALOG=ON
-- The C compiler identification is GNU 11.5.0
-- The CXX compiler identification is GNU 11.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- pg_config.h => /home/gpadmin/cloudberry/contrib/pax_storage/../../src/include/pg_config.h
-- Looking for USE_ASSERT_CHECKING
-- Looking for USE_ASSERT_CHECKING - found
-- Looking for USE_LZ4
-- Looking for USE_LZ4 - found
-- PAX BUILD TYPE: Debug
-- Looking for USE_VECTORIZATION
-- Looking for USE_VECTORIZATION - not found
-- Build pax without vectorization support
-- Found Protobuf: /usr/lib64/libprotobuf.so (found version "3.14.0") 
-- Found Protobuf: /usr/lib64/libprotobuf.so (found suitable version "3.14.0", minimum required is "3.5.0") 
-- Checking for module 'libzstd'
--   Found libzstd, version 1.5.5
-- tabulate version: v1.5.0
-- Using C++11
-- Found Threads: TRUE  
-- Configuring done (1.5s)
-- Generating done (0.1s)
-- Build files have been written to: /home/gpadmin/cloudberry/contrib/pax_storage/build
cd build && make -j8
make[2]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[2]: warning: -j8 forced in submake: resetting jobserver mode.
make[3]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  0%] Building C object src/cpp/CMakeFiles/generate_sql_script_program.dir/__/__/tools/gen_sql.c.o
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  0%] Building CXX object src/cpp/contrib/googletest/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
[  0%] Built target generate_protobuf
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  0%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/orc_proto.pb.cc.o
[  0%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/pax.pb.cc.o
[  1%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column_traits.cc.o
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/micro_partition_stats.pb.cc.o
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column.cc.o
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_compress.cc.o
[  2%] Linking C executable generate_sql_script_program
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  2%] Built target generate_sql_script_program
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  2%] dynamically generate sql script file
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  2%] Built target create_sql_script
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  2%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/micro_partition_stats.pb.cc.o
[  3%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/pax.pb.cc.o
[  3%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/orc_proto.pb.cc.o
[  4%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_columns.cc.o
[  4%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_column_traits.cc.o
[  4%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding_utils.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_column.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_compress.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_columns.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding_non_fixed_column.cc.o
[  6%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding_utils.cc.o
[  7%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding_column.cc.o
[  7%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_dict_encoding.cc.o
[  7%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_decoding.cc.o
[  7%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding_non_fixed_column.cc.o
[  7%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding_column.cc.o
[  8%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_encoding.cc.o
[  9%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_dict_encoding.cc.o
[  9%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_decoding.cc.o
[  9%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_delta_encoding.cc.o
[  9%] Linking CXX static library ../../../../../lib/libgtest.a
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  9%] Built target gtest
[  9%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_encoding.cc.o
[  9%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_rlev2_decoding.cc.o
[ 10%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_rlev2_encoding.cc.o
[ 10%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_column.cc.o
[ 10%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_bitpacked_column.cc.o
[ 11%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_delta_encoding.cc.o
[ 11%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_rlev2_decoding.cc.o
[ 12%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_bpchar_column.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 13%] Building CXX object src/cpp/contrib/googletest/googlemock/CMakeFiles/gmock.dir/src/gmock-all.cc.o
[ 13%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_encoding_column.cc.o
[ 13%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_vec_numeric_column.cc.o
[ 14%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/oper/pax_oper_udf.cc.o
[ 14%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_rlev2_encoding.cc.o
[ 14%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_filter.cc.o
[ 14%] Linking CXX static library ../../../../../lib/libgmock.a
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 15%] Building CXX object src/cpp/contrib/googletest/googletest/CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 15%] Built target gmock
[ 16%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_bitpacked_column.cc.o
[ 16%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_row_filter.cc.o
[ 16%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_bpchar_column.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 16%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/micro_partition_stats.pb.cc.o
[ 16%] Linking CXX static library ../../../../../lib/libgtest_main.a
[ 17%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/pax.pb.cc.o
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 17%] Built target gtest_main
[ 18%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_sparse_filter.cc.o
[ 18%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_column.cc.o
[ 18%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_sparse_pg_path.cc.o
[ 18%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/filter/pax_sparse_vec_path.cc.o
[ 19%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/oper/pax_oper.cc.o
[ 19%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/orc_proto.pb.cc.o
[ 19%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/oper/pax_stats.cc.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 19%] Building CXX object src/cpp/contrib/googletest/googlemock/CMakeFiles/gmock_main.dir/src/gmock_main.cc.o
[ 19%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/file_system.cc.o
[ 20%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/local_file_system.cc.o
[ 20%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_column_traits.cc.o
[ 21%] Linking CXX static library ../../../../../lib/libgmock_main.a
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[ 21%] Built target gmock_main
[ 22%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_encoding_column.cc.o
[ 22%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition.cc.o
[ 22%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_file_factory.cc.o
[ 23%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_column.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_metadata.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_compress.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_row_filter_reader.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/columns/pax_vec_numeric_column.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_stats.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_columns.cc.o
[ 24%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/oper/pax_oper_udf.cc.o
[ 25%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding_utils.cc.o
[ 26%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_stats_updater.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_filter.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_row_filter.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding_non_fixed_column.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_sparse_filter.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/micro_partition_udf.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding_column.cc.o
[ 27%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_dump_reader.cpp.o
[ 28%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_sparse_pg_path.cc.o
[ 29%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_dict_encoding.cc.o
[ 30%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_format_reader.cc.o
[ 30%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_decoding.cc.o
[ 30%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/filter/pax_sparse_vec_path.cc.o
[ 30%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_encoding.cc.o
[ 30%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/oper/pax_oper.cc.o
[ 31%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/oper/pax_stats.cc.o
[ 32%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_delta_encoding.cc.o
[ 32%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_rlev2_decoding.cc.o
[ 32%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_group.cc.o
[ 32%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_vec_group.cc.o
[ 32%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/file_system.cc.o
[ 33%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_reader.cc.o
[ 33%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/local_file_system.cc.o
[ 33%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_rlev2_encoding.cc.o
[ 34%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition.cc.o
[ 34%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_file_factory.cc.o
[ 35%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_bitpacked_column.cc.o
[ 35%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_type.cc.o
[ 35%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_metadata.cc.o
[ 35%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_bpchar_column.cc.o
[ 35%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/orc/orc_writer.cc.o
[ 36%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_row_filter_reader.cc.o
[ 36%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_column.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_encoding_column.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/columns/pax_vec_numeric_column.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/oper/pax_oper_udf.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/pax_buffer.cc.o
[ 37%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_stats.cc.o
[ 38%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/protobuf_stream.cc.o
[ 39%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_filter.cc.o
[ 39%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/toast/pax_toast.cc.o
[ 39%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_row_filter.cc.o
[ 39%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_stats_updater.cc.o
[ 40%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_udf.cc.o
[ 40%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_dump_reader.cpp.o
[ 40%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_sparse_filter.cc.o
[ 40%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_format_reader.cc.o
[ 41%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_group.cc.o
[ 42%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_sparse_pg_path.cc.o
[ 42%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/wal/pax_wal.cc.o
[ 42%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/filter/pax_sparse_vec_path.cc.o
[ 42%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/oper/pax_oper.cc.o
[ 43%] Building C object src/cpp/CMakeFiles/paxformat.dir/storage/wal/paxc_desc.c.o
[ 43%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/wal/paxc_wal.cc.o
[ 43%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/clustering.cc.o
[ 44%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/sorter_tuple.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/oper/pax_stats.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/sorter_index.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_vec_group.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_reader.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/zorder_clustering.cc.o
[ 45%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/file_system.cc.o
[ 46%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_type.cc.o
[ 47%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/index_clustering.cc.o
[ 47%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/local_file_system.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/lexical_clustering.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/orc/orc_writer.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/clustering/zorder_utils.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_file_factory.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/pax_buffer.cc.o
[ 48%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_metadata.cc.o
[ 49%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/exceptions/CException.cc.o
[ 50%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_row_filter_reader.cc.o
[ 50%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_stats.cc.o
[ 51%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/pax_itemptr.cc.o
[ 51%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/bitmap.cc.o
[ 51%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/bloomfilter.cc.o
[ 51%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/proto/protobuf_stream.cc.o
[ 51%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_stats_updater.cc.o
[ 51%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/pax.cc.o
[ 52%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/byte_buffer.cc.o
[ 53%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/paxc_smgr.cc.o
[ 54%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_udf.cc.o
[ 54%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/fast_io.cc.o
[ 54%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_dump_reader.cpp.o
[ 54%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/comm/guc.cc.o
[ 54%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_format_reader.cc.o
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/comm/fast_io.cc: In static member function ‘static bool pax::IOUringFastIO::available()’:
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/comm/fast_io.cc:48:24: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   48 |   if (support_io_uring == -1) return false;
      |       ~~~~~~~~~~~~~~~~~^~~~~
[ 54%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/toast/pax_toast.cc.o
cc1plus: all warnings being treated as errors
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:944: src/cpp/CMakeFiles/paxformat.dir/comm/fast_io.cc.o] Error 1
make[4]: *** Waiting for unfinished jobs....
[ 54%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/strategy.cc.o
[ 55%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/wal/pax_wal.cc.o
[ 55%] Building C object src/cpp/CMakeFiles/pax.dir/storage/wal/paxc_desc.c.o
[ 56%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_group.cc.o
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[3]: *** [CMakeFiles/Makefile2:205: src/cpp/CMakeFiles/paxformat.dir/all] Error 2
make[3]: *** Waiting for unfinished jobs....
[ 56%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/wal/paxc_wal.cc.o
[ 56%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_vec_group.cc.o
[ 56%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_reader.cc.o
[ 57%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_type.cc.o
[ 57%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/orc/orc_writer.cc.o
[ 58%] Building CXX object src/cpp/CMakeFiles/pax.dir/storage/micro_partition_iterator.cc.o
[ 58%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/clustering.cc.o
[ 58%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/sorter_tuple.cc.o
[ 58%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/pax_buffer.cc.o
[ 59%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/sorter_index.cc.o
[ 59%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/zorder_clustering.cc.o
[ 60%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/pax_itemptr.cc.o
[ 60%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/proto/protobuf_stream.cc.o
[ 60%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/index_clustering.cc.o
[ 61%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/lexical_clustering.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/pax.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/paxc_smgr.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/pax_clustering_reader.cc.o
[ 62%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/pax_clustering_writer.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/pax.dir/clustering/zorder_utils.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/toast/pax_toast.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/strategy.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/pax.dir/exceptions/CException.cc.o
[ 63%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/paxc_rel_options.cc.o
[ 64%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/wal/pax_wal.cc.o
[ 65%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_access_handle.cc.o
[ 65%] Building C object src/cpp/CMakeFiles/test_main.dir/storage/wal/paxc_desc.c.o
[ 65%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_access_method_internal.cc.o
[ 65%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_deleter.cc.o
[ 65%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/wal/paxc_wal.cc.o
[ 66%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_dml_state.cc.o
[ 66%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_inserter.cc.o
[ 67%] Building CXX object src/cpp/CMakeFiles/test_main.dir/storage/micro_partition_iterator.cc.o
[ 67%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_table_cluster.cc.o
[ 68%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_updater.cc.o
[ 68%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/clustering.cc.o
[ 68%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/sorter_tuple.cc.o
[ 69%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/sorter_index.cc.o
[ 69%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/zorder_clustering.cc.o
[ 69%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/index_clustering.cc.o
[ 70%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/lexical_clustering.cc.o
[ 70%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/pax_clustering_reader.cc.o
[ 70%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/pax_clustering_writer.cc.o
[ 71%] Building CXX object src/cpp/CMakeFiles/test_main.dir/clustering/zorder_utils.cc.o
[ 71%] Building CXX object src/cpp/CMakeFiles/test_main.dir/exceptions/CException.cc.o
[ 71%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/paxc_rel_options.cc.o
[ 71%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_visimap.cc.o
[ 72%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_access_handle.cc.o
[ 72%] Building CXX object src/cpp/CMakeFiles/pax.dir/access/pax_scanner.cc.o
[ 73%] Building CXX object src/cpp/CMakeFiles/pax.dir/comm/bitmap.cc.o
[ 73%] Building CXX object src/cpp/CMakeFiles/pax.dir/comm/bloomfilter.cc.o
[ 73%] Building CXX object src/cpp/CMakeFiles/pax.dir/comm/byte_buffer.cc.o
[ 74%] Building CXX object src/cpp/CMakeFiles/pax.dir/comm/fast_io.cc.o
[ 74%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_access_method_internal.cc.o
[ 74%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_deleter.cc.o
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/comm/fast_io.cc: In static member function ‘static bool pax::IOUringFastIO::available()’:
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/comm/fast_io.cc:48:24: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   48 |   if (support_io_uring == -1) return false;
      |       ~~~~~~~~~~~~~~~~~^~~~~
cc1plus: all warnings being treated as errors
make[4]: *** [src/cpp/CMakeFiles/pax.dir/build.make:1182: src/cpp/CMakeFiles/pax.dir/comm/fast_io.cc.o] Error 1
make[4]: *** Waiting for unfinished jobs....
[ 75%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_dml_state.cc.o
[ 75%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_inserter.cc.o
[ 75%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_table_cluster.cc.o
[ 76%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_updater.cc.o
[ 76%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_visimap.cc.o
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[3]: *** [CMakeFiles/Makefile2:310: src/cpp/CMakeFiles/pax.dir/all] Error 2
[ 76%] Building CXX object src/cpp/CMakeFiles/test_main.dir/access/pax_scanner.cc.o
[ 77%] Building CXX object src/cpp/CMakeFiles/test_main.dir/comm/bitmap.cc.o
[ 77%] Building CXX object src/cpp/CMakeFiles/test_main.dir/comm/bloomfilter.cc.o
[ 77%] Building CXX object src/cpp/CMakeFiles/test_main.dir/comm/byte_buffer.cc.o
[ 78%] Building CXX object src/cpp/CMakeFiles/test_main.dir/comm/fast_io.cc.o
[ 78%] Building CXX object src/cpp/CMakeFiles/test_main.dir/comm/guc.cc.o
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/comm/fast_io.cc: In static member function ‘static bool pax::IOUringFastIO::available()’:
/home/gpadmin/cloudberry/contrib/pax_storage/src/cpp/comm/fast_io.cc:48:24: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   48 |   if (support_io_uring == -1) return false;
      |       ~~~~~~~~~~~~~~~~~^~~~~
cc1plus: all warnings being treated as errors
make[4]: *** [src/cpp/CMakeFiles/test_main.dir/build.make:1182: src/cpp/CMakeFiles/test_main.dir/comm/fast_io.cc.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[3]: *** [CMakeFiles/Makefile2:339: src/cpp/CMakeFiles/test_main.dir/all] Error 2
make[3]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[2]: *** [Makefile:136: all] Error 2
make[2]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[1]: *** [Makefile:69: build] Error 2
make[1]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage'
make: *** [GNUmakefile:32: all] Error 2
make: Leaving directory '/home/gpadmin/cloudberry'
```

### What you think should happen instead

Should work well.

### How to reproduce

Following the docs here: https://cloudberry.apache.org/docs/next/deployment/quick-build

### Operating System

Rocky Linux 8+9

### Anything else

OS Info

1. Rocky Linux 8

```
[gpadmin@cdw cloudberry]$ gcc --version
gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-28)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[gpadmin@cdw cloudberry]$ g++ --version
g++ (GCC) 8.5.0 20210514 (Red Hat 8.5.0-28)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[gpadmin@cdw cloudberry]$ cat /etc/os-release 
NAME="Rocky Linux"
VERSION="8.10 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.10"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.10 (Green Obsidian)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2029-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-8"
ROCKY_SUPPORT_PRODUCT_VERSION="8.10"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.10"
```

2. Rocky Linux 9

```
[gpadmin@cdw cloudberry]$ gcc --version
gcc (GCC) 11.5.0 20240719 (Red Hat 11.5.0-11)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[gpadmin@cdw cloudberry]$ g++ --version
g++ (GCC) 11.5.0 20240719 (Red Hat 11.5.0-11)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[gpadmin@cdw cloudberry]$ cat /etc/os-release 
NAME="Rocky Linux"
VERSION="9.7 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.7"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.7 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
VENDOR_NAME="RESF"
VENDOR_URL="https://resf.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.7"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.7"
```


### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/cloudberry/blob/main/CODE_OF_CONDUCT.md).

--- a/contrib/pax_storage/src/cpp/comm/fast_io.cc
+++ b/contrib/pax_storage/src/cpp/comm/fast_io.cc
@@ -26,7 +26,7 @@
  */
 
 #include "fast_io.h"
-
+#include <stdint.h>
 #include <unistd.h>  // for pread
 
 // uring_likely may not be defined in older liburing versions
@@ -42,7 +42,7 @@ namespace pax
 {
 
 bool IOUringFastIO::available() {
-  static char support_io_uring = 0;
+  static int8_t support_io_uring = 0;
 
   if (support_io_uring == 1) return true;
   if (support_io_uring == -1) return false;

